app/main/views/questionnaire.py

```diff
@@ -81,18 +80,27 @@ def get_thank_you(eq_id, form_type, period_id, collection_id):
     if not same_survey(eq_id, form_type, period_id, collection_id):
         return redirect("/information/multiple-surveys")
 
-    g.questionnaire_manager.go_to('thank-you')
     thank_you_page = render_page('thank-you', True)
     # Delete user data on request of thank you page.
     delete_user_data()
     return thank_you_page
 
 
-@questionnaire_blueprint.route('summary', methods=["GET"])
+@questionnaire_blueprint.route('submit-answers', methods=["POST"])
 @login_required
-def get_summary(eq_id, form_type, period_id, collection_id):
-    g.questionnaire_manager.go_to('summary')
-    return render_page('summary', True)
+def submit_answers(eq_id, form_type, period_id, collection_id):
+    answers = get_answers(current_user)
+
+    # check that all the answers we have are valid before submitting the data
+    is_valid, invalid_location = g.questionnaire_manager.validate_all_answers()
+
+    if is_valid:
+        submitter = SubmitterFactory.get_submitter()
+        submitter.send_answers(current_user, get_metadata(current_user), g.questionnaire_manager._schema, answers)
```

Comment:
`current_user` isn't used by the submitter. Also `_schema` is a 'private' property of questionnaire manager, we should either rename the property `schema` or create an accessor.